jerry-core/ecma/base/ecma-gc.c

```diff
@@ -1245,6 +1262,23 @@ ecma_gc_free_executable_object (ecma_object_t *object_p) /**< object */
     {
       context_top_p[-1] &= (uint32_t) ~VM_CONTEXT_HAS_LEX_ENV;
 
+      if (VM_CONTEXT_HAS_VARIABLE_LENGTH (VM_GET_CONTEXT_TYPE (context_top_p[-1])))
+      {
+        int32_t context_size = (int32_t) VM_GET_CONTEXT_END (context_top_p[-1]);
+
+        for (int32_t offset = 2; offset <= context_size; offset++)
+        {
+          if (ecma_is_value_object (context_top_p[-offset]))
+          {
+            context_top_p[offset] = ECMA_VALUE_UNDEFINED;
```

Comment:
I think this is incorrect. The code below calls `vm_stack_context_abort` after the object values are cleared. Maybe freeing the non-objects could work.

Reply:
Good catch. Moreover only a `vm_stack_context_abort` is enough here.